ui/src/components/App.js

```diff
@@ -21,65 +18,105 @@ import HelpContainer from '../containers/HelpContainer';
 import Main from './Main';
 import LoadingScreen from '../components/LoadingScreen/LoadingScreen';
 
-async function requireAuth() {
-  await store.dispatch(getLoginInfo());
-  const { login } = store.getState();
+import { store } from '../store';
+import { serverIO } from '../serverIO';
 
-  if (login.loggedIn) {
-    await store.dispatch(startSession(login.user.inControl));
-  }
+import { getLoginInfo } from '../actions/login';
+import { getInitialState, applicationFetched } from '../actions/general';
 
-  if (login.loggedIn && !serverIO.initialized) {
-    serverIO.listen(store);
+export async function requireAuth() {
+  try {
+    store.dispatch(applicationFetched(false));
+    await store.dispatch(getLoginInfo());
+  } catch {
+    store.dispatch(applicationFetched(true));
+    return;
   }
+
+  await store.dispatch(getInitialState());
+  serverIO.listen(store);
 }
 
 function PrivateOutlet() {
   const loggedIn = useSelector((state) => state.login.loggedIn);
   const location = useLocation();
-  requireAuth();
   return loggedIn ? (
     <Outlet />
   ) : (
     <Navigate to="/login" state={{ from: location }} replace />
   );
 }
 
+const router = createBrowserRouter([
+  {
+    path: '/login',
+    element: <LoginContainer />,
+  },
+  {
+    path: '/',
+    element: <PrivateOutlet />,
+    loader: () => {
+      requireAuth();
+      return true;
+    },
+    children: [
+      {
+        path: '',
+        element: <Main />,
+        children: [
+          {
+            index: true,
+            element: <SampleViewContainer />,
+          },
+          {
+            path: 'samplegrid',
+            element: <SampleListViewContainer />,
+          },
+          {
+            path: 'datacollection',
+            element: <SampleViewContainer />,
+          },
+          {
+            path: 'equipment',
+            element: <EquipmentContainer />,
+          },
+          {
+            path: 'logging',
+            element: <LoggerContainer />,
+          },
+          {
+            path: 'remoteaccess',
+            element: <RemoteAccessContainer />,
+          },
+          {
+            path: 'help',
+            element: <HelpContainer />,
+          },
+        ],
+      },
+    ],
+  },
+]);
+
 function App(props) {
-  const { loggedIn } = props;
+  const { applicationFetched } = props;
 
   useEffect(() => {
     requireAuth();
```

Comment:
Might be cleaner to inline the function in here and use `mapDispatchToProps` to bind the various action creators.